include/ginkgo/core/base/executor.hpp

```diff
@@ -755,6 +817,12 @@ namespace detail {
 
 template <typename ConcreteExecutor>
 class ExecutorBase : public Executor {
+#define GKO_DECLARE_EXECUTOR_FRIEND(_type, ...) friend class _type
```

Comment:
You have defined this macro twice. Maybe you can move it to a common namespace and declare it only once ?